### PR TITLE
fix: only setFailed when actor not null

### PR DESCRIPTION
**Pull request by Roiocam**
continue #1385, we allow empty actorRef now.

### CODE REVIEW

File: actor/src/main/scala/org/apache/pekko/actor/ActorCell.scala [hunk 1]
```diff
@@ -661,8 +661,7 @@ private[pekko] class ActorCell(
         Thread.currentThread().interrupt()
         throw ActorInitializationException(self, "interruption during creation", e)
       case NonFatal(e) =>
-        if (actor == null) setFailed(system.deadLetters)
-        else setFailed(actor.self)
+        if (actor != null) setFailed(actor.self)
```
Comment (He-Pin) [inline actor/src/main/scala/org/apache/pekko/actor/ActorCell.scala:664]: Use ne 

A test is needed too

Comment (Roiocam) [reply actor/src/main/scala/org/apache/pekko/actor/ActorCell.scala]: resolves

**Review by pjfanning — APPROVED** [review]
lgtm

can you backport this to 1.0.x when you merge it?

**Comment by Roiocam** [comment]
tests passed in https://github.com/apache/pekko/actions/runs/9987454134

**Review by He-Pin — APPROVED** [review]

**Comment by Roiocam** [comment]
@sadekmunawar would your take a look?

**Review by sadekmunawar — APPROVED** [review]
Looks good

**Comment by Roiocam** [comment]
> Looks good

Thanks for the review @sadekmunawar

**Pull request merged**